embedded/src/main/java/org/jboss/as/embedded/StandaloneServer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2010, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.embedded;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import javax.naming.Context;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+
+/**
+ * The standalone server interface.
+ *
+ * @author Thomas.Diesler@jboss.com
+ * @since 17-Nov-2010
+ */
+public interface StandaloneServer {
+
+    // TODO: use a DeploymentPlan
+    @Deprecated
+    void deploy(File file) throws IOException, ExecutionException, InterruptedException;
+
+    // TODO: use a DeploymentPlan
+    @Deprecated
+    void undeploy(File file) throws ExecutionException, InterruptedException;
```

Comment:
I didn't really intend this to be a new impl, just a port of the existing one. I gave it a new module name just so it could temporarily co-exist with the one in WF full.

This "port vs ???" question is one of the areas to discuss about this work though. If it's *not* just a port, sure, deprecated methods should not come in. If it is just a port, getting rid of them is a valid question too, but it's a separate issue.